src/sttfa/export/agda.ml

```diff
@@ -0,0 +1,175 @@
+module D = Core.Deps
+open Ast
+
+let sys = "agda"
+
+let cur_md = ref ""
+
+(* no files in arith_fermat are using them but we still might get new files using them in the future *)
+let forbidden_id = ref ["abstract";"consructor";"data";"do";"eta-equality";"field";"forall";"hiding";
+                        "import";"in";"inductive";"infix";"infixl";"infixr";"instance";"let";"macro";
+                        "module";"mutual";"no-eta-equality";"open";"overlap";"pattern";"postulate";
+                        "primitive";"private";"public";"quote";"quoteContext";"quoteGoal";"quoteTerm";
+                        "record";"renaming";"rexrite";"Set";"syntax";"tactic";"unquote";"unquoteDecl";
+                        "unquoteDef";"using";"variable";"where";"with"]
+
+(* when _ used as wildcard, leave it
+  otherwise, change it to :: (- is forbidden because it may lead to inline comments) 
+  because i would be interpreted as mix-fix operator *)
+let sanitize id =
+  if id = "_" 
+  then id
+  else 
+    if List.mem id !forbidden_id
+    then 
+      "dk^"^id
+    else
+      let regexp = Str.regexp "_" in
+      Str.global_replace regexp "::" id
```

Comment:
_ changed to ::
I tried with ^ but it wa stoo ugly and - can't be used or it could be interpreted as a comment